nf_core/pipeline-template/.github/workflows/ci.yml

```diff
@@ -39,21 +43,10 @@ jobs:
       - name: Disk space cleanup
         uses: jlumbroso/free-disk-space@54081f138730dfa15788a46383842cd2f914a1be # v1.3.1
 
-      - name: Run pipeline with test data (docker)
+      - name: "Run pipeline with test data {% raw %}${{ matrix.NXF_VER }}{% endraw %} | {% raw %}${{ matrix.engine }}{% endraw %}"
         # TODO nf-core: You can customise CI pipeline run tests as required
         # For example: adding multiple test runs with different parameters
         # Remember that you can parallelise this by using strategy.matrix
         run: |
-          nextflow run ${GITHUB_WORKSPACE} -profile test,docker --outdir ./results
-
-      - name: Run pipeline with test data (singularity)
-        # TODO nf-core: You can customise CI pipeline run tests as required
-        run: |
-          nextflow run ${GITHUB_WORKSPACE} -profile test,singularity --outdir ./results
-        if: "{% raw %}${{ github.base_ref == 'master' }}{% endraw %}"
-
-      - name: Run pipeline with test data (conda)
-        # TODO nf-core: You can customise CI pipeline run tests as required
-        run: |
-          nextflow run ${GITHUB_WORKSPACE} -profile test,conda --outdir ./results
-        if: "{% raw %}${{ github.base_ref == 'master' }}{% endraw %}"
+          nextflow run ${GITHUB_WORKSPACE} -profile test,{% raw %}${{ matrix.engine }}{% endraw %} --outdir ./results
+          if: "!{% raw %}${{ github.base_ref == 'master' }}{% endraw %} || {% raw %}${{ matrix.engine == 'docker' }}{% endraw %}"
```

Comment:
I think the `if` statement has one indentation more, it should be at the same level as `run`